File: Medium/3-Longest-Substring-Without-Repeating-Characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        curr_window = set()
        max_len, l = 0, 0

        for i, c in enumerate(s):
            while c in curr_window:
                curr_window.remove(s[l])
                l += 1

            curr_window.add(c)
            max_len = max(max_len, i - l + 1)
            
        return max_len

File: Medium/3-Longest-Substring-Without-Repeating-Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        String longest = "";
        int ans = 0;
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            while (longest.indexOf(c) != -1) {
                longest = longest.substring(1);
            }
            longest += c;
            ans = Math.max(ans, longest.length());
        }
        
        return ans;
    }
}
